sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/XmlExternalEntityShouldNotBeParsedTest.cs

```diff
@@ -99,8 +99,8 @@ public void XmlExternalEntityShouldNotBeParsed_NoCrashOnExternalParameterUse()
             Verifier.VerifyAnalyzer(
                 new[]
                 {
-                    @"TestCases\XmlExternalEntityShouldNotBeParsed_XmlReader_ParameterProvider.cs",
-                    @"TestCases\XmlExternalEntityShouldNotBeParsed_XmlReader_ExternalParameter.cs"
+                    @"TestCases\XmlExternalEntityShouldNotBeParsed_XmlReader_ExternalParameter.cs",
+                    @"TestCases\XmlExternalEntityShouldNotBeParsed_XmlReader_ParameterProvider.cs"
```

Comment:
As far as I can tell we have a bug in the `DiagnosticVerifier`. It seems to check the expected issues only in the first file.

Reply:
Not sure I understand this comment related to this change. And if we have a bug, do we have ticket to track it ?

Reply:
I took a better look this morning and I don't think that it's a bug anymore, just a partial implementation :).

Currently the location does dot know about the file: https://github.com/SonarSource/sonar-dotnet/blob/master/sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/IIssueLocation.cs

and the **DiagnosticVerifier** will check only in the first file for issue locations: https://github.com/SonarSource/sonar-dotnet/blob/master/sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs#L56

Please note that the first syntax tree (the one that is skipped) is for "ExtraEmptyFile.g.cs" which is added to the project by default: https://github.com/SonarSource/sonar-dotnet/blob/master/sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/SolutionBuilder.cs#L81

Probably checking for issues in multiple files was never needed.

Reply:
Ok. Makes sense now.
I think it might be worth it to add a comment for method declaration `public static void VerifyAnalyzer(IEnumerable<string> paths,` making it explicit that only the 1st file will be checked by the verifier. After looking in the code, I indeed remember now that these extra paths are used for additional file dependency. Since it's not something that we use that often, making it clear might help in the future